sklearn/compose/tests/test_column_transformer.py

```diff
@@ -2229,3 +2255,18 @@ def test_remainder_set_output():
     ct.set_output(transform="default")
     out = ct.fit_transform(df)
     assert isinstance(out, np.ndarray)
+
+
+def test_dataframe_interchange_errors():
+    """DataFrame interchange requires the ColumnTranformer to be fitted on dataframe."""
+    pl = pytest.importorskip("polars")
+
+    X_train = np.array([[0, 1], [2, 4]]).T
+    X_test = pl.DataFrame({"a": [1, 2, 3], "b": [1, 2, 3]})
+
+    ct = make_column_transformer((Trans(), [0, 1]))
+    ct.fit(X_train)
+
+    msg = "Using the dataframe protocol requires fitting on dataframes."
+    with pytest.raises(ValueError, match=msg):
+        ct.transform(X_test)
```

Comment:
What about fitting on polars and transforming on pandas, or vice-versa?

Reply:
That works without any issue. I added a test here: [`e4e8249` (#26683)](https://github.com/scikit-learn/scikit-learn/pull/26683/commits/e4e82493e3fbdd7dcff4b19b66f3a02827f55fe3)

Reply:
Thanks.